starport/services/network/join.go

```diff
@@ -170,26 +181,42 @@ func (n Network) sendValidatorRequest(
 }
 
 // hasValidator verify if the validator already exist into the SPN store
-func (n Network) hasValidator(ctx context.Context, launchID uint64, address string) bool {
+func (n Network) hasValidator(ctx context.Context, launchID uint64, address string) (bool, error) {
 	_, err := launchtypes.NewQueryClient(n.cosmos.Context).GenesisValidator(ctx, &launchtypes.QueryGetGenesisValidatorRequest{
 		LaunchID: launchID,
 		Address:  address,
 	})
-	return err == nil
+	err = sdkerror.Unwrap(err)
+	if err == sdkerror.ErrInvalidRequest {
+		return false, nil
+	} else if err != nil {
+		return false, err
+	}
+	return true, nil
 }
 
 // hasAccount verify if the account already exist into the SPN store
-func (n Network) hasAccount(ctx context.Context, launchID uint64, address string) bool {
+func (n Network) hasAccount(ctx context.Context, launchID uint64, address string) (bool, error) {
 	_, err := launchtypes.NewQueryClient(n.cosmos.Context).VestingAccount(ctx, &launchtypes.QueryGetVestingAccountRequest{
 		LaunchID: launchID,
 		Address:  address,
 	})
-	if err == nil {
-		return true
+	err = sdkerror.Unwrap(err)
+	if err != nil && err != sdkerror.ErrInvalidRequest {
+		return false, err
+	} else if err == nil {
+		return true, nil
 	}
+
 	_, err = launchtypes.NewQueryClient(n.cosmos.Context).GenesisAccount(ctx, &launchtypes.QueryGetGenesisAccountRequest{
 		LaunchID: launchID,
 		Address:  address,
 	})
-	return err == nil
+	err = sdkerror.Unwrap(err)
+	if err != nil && err != sdkerror.ErrInvalidRequest {
+		return false, err
+	} else if err == nil {
+		return true, nil
+	}
```

Comment:
```suggestion
	if err == nil {
		return true, nil
	} else if err != sdkerror.ErrInvalidRequest {
		return false, err
	}
```
I think this way it will be more explicit.